From 280f7fcc36bdaa3ad5fe49ee34e4e4cf3cd62980 Mon Sep 17 00:00:00 2001
From: Weiwei Wang <weiwei.wang@windriver.com>
Date: Tue, 3 Nov 2009 10:33:42 +0800
Subject: [PATCH] x86: delete redundant vbiVioapicIntUnmask

It is meaningless to call vbiVioapicIntUnmask in ipi
handlers, so delete them.

Signed-off-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 arch/x86/kernel/vbi/wrhv.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/arch/x86/kernel/vbi/wrhv.c b/arch/x86/kernel/vbi/wrhv.c
index 96d2f5d..ab2b2f1 100644
--- a/arch/x86/kernel/vbi/wrhv.c
+++ b/arch/x86/kernel/vbi/wrhv.c
@@ -409,7 +409,6 @@ irqreturn_t wrhv_ipi_func_call_single_handler(int irq, void *dev_id)
 	generic_smp_call_function_single_interrupt();
 	__get_cpu_var(irq_stat).irq_call_count++;
 	irq_exit();
-	vbiVioapicIntUnmask(irq);
 	return IRQ_HANDLED;
 }
 
@@ -419,7 +418,6 @@ irqreturn_t wrhv_ipi_func_call_handler(int irq, void *dev_id)
 	generic_smp_call_function_interrupt();
 	__get_cpu_var(irq_stat).irq_call_count++;
 	irq_exit();
-	vbiVioapicIntUnmask(irq);
 	return IRQ_HANDLED;
 }
 
@@ -470,14 +468,12 @@ out:
 	put_cpu_no_resched();
 	__get_cpu_var(irq_stat).irq_tlb_count++;
 
-	vbiVioapicIntUnmask(irq);
 	return IRQ_HANDLED;
 }
 
 irqreturn_t wrhv_ipi_resched_handler(int irq, void *dev_id)
 {
 	__get_cpu_var(irq_stat).irq_resched_count++;
-	vbiVioapicIntUnmask(irq);
 	return IRQ_HANDLED;
 }
 #endif
-- 
1.6.5.2

